cursor.c

```diff
@@ -277,6 +285,18 @@ PHP_METHOD(MongoCursor, hasNext)
 	if (!php_mongo_get_more(cursor TSRMLS_CC)) {
 		RETURN_FALSE;
 	}
+
```

Comment:
Shouldn't we do this check *after* `php_mongo_handle_error()`? I think we'd want to handle a possible error (which means throwing an exception) immediately after getmore. As-is, we may return `false` and skip a chance to throw an exception from a getmore.

Also, after the `php_mongo_handle_error()` check we do:

```c
if (have_error_flags(cursor)) {
	RETURN_TRUE;
}
```

What is the reasoning behind returning true if `MONGO_OP_REPLY_ERROR_FLAGS` is set within the cursor's flags? That actually seems like dead code because `php_mongo_handle_error()` will throw exceptions for `(cursor->flag & MONGO_OP_REPLY_ERROR_FLAGS)`.